contracts/lbp/LBPManager.sol

```diff
@@ -0,0 +1,290 @@
+
+    modifier onlyAdmin() {
+        require(msg.sender == admin, "LBPManager: caller should be admin");
+        _;
+    }
+
+    /**
+     * @dev                             Transfer admin rights.
+     * @param _newAdmin                 Address of the new admin.
+     */
+    function transferAdminRights(address _newAdmin) external onlyAdmin {
+        require(
+            _newAdmin != address(0),
+            "LBPManager: new admin can not be zero"
+        );
+        admin = _newAdmin;
+    }
+
+    /**
+     * @dev                             Initialize LBPManager.
+     * @param _LBPFactory               LBP factory address.
+     * @param _beneficiary              The address that receives the _primeDaoFeePercentage.
+     * @param _name                     Name of the LBP.
+     * @param _symbol                   Symbol of the LBP.
+     * @param _tokenList                Numerically sorted array (ascending) containing two addresses:
+                                            - The address of the project token being distributed.
+                                            - The address of the funding token being exchanged for the project token.
+     * @param _amounts                  Sorted array to match the _tokenList, containing two parameters:
+                                            - The amounts of project token to be added as liquidity to the LBP.
+                                            - The amounts of funding token to be added as liquidity to the LBP.
+     * @param _weights                  Sorted array to match the _tokenList, containing two parametes:
+                                            - The start weight for the project token in the LBP.
+                                            - The start weight for the funding token in the LBP.
+     * @param _startTimeEndTime         Array containing two parameters:
+                                            - Start time for the LBP.
+                                            - End time for the LBP.
+     * @param _endWeights               Sorted array to match the _tokenList, containing two parametes:
+                                            - The end weight for the project token in the LBP.
+                                            - The end weight for the funding token in the LBP.
+     * @param _swapFeePercentage        Percentage of fee paid for every swap in the LBP.
+     * @param _primeDaoFeePercentage    Percentage of fee paid to PrimeDao for providing the service of the LBP Manager.
+     */
+    function initializeLBP(
+        address _LBPFactory,
+        address _beneficiary,
+        string memory _name,
+        string memory _symbol,
+        IERC20[] memory _tokenList,
+        uint256[] memory _amounts,
+        uint256[] memory _weights,
+        uint256[] memory _startTimeEndTime,
+        uint256[] memory _endWeights,
+        uint256 _swapFeePercentage,
+        uint256 _primeDaoFeePercentage
```

Comment:
Hmm, I am not clear here, as I understand we need to set fee to 0(according to issue #112 ), but here the fee is configurable? Which exact fee do we need to set to 0?

Reply:
We kept track of all the changes that got decided on in the Miro board as well as a spec doc. I did not change the issue to reflect the constant changes and when creating this final PR, I just added all the issues that were related to this not thinking it through that the reviewer would, of course, use the issues as reference.

In the case of the `swapFeePercentage`, Balancer did not allow zero percentage so we first used the lowest possible, set it as a constant in the `LBPManagerFactory`, passing it to the manager on deploying it. We ended up with a decision where a swap fee should be used and passed from the dapp every time a LBP gets created through the wizard. That is the implementation you see here.

I updated the issues to reflect the changes. My apologies for it being a bit messy in this way.

Reply:
Hmm, so `_primeDaoFeePercentage` should not be changed after the contract was deployed?

Reply:
Yes indeed, the fee for the service will be set in the wizard when creating the LBP. This fee will stay the same for that LBP and LBPManager instance